src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj

```diff
@@ -51,17 +53,21 @@
   </ItemGroup>
 
   <PropertyGroup>
-    <!--
-      We check in the Release / Production build of blazor.*.js, but not the Debug builds. Consequently the former is always available to embed, the latter is only available
-      if Web.JS was built locally. Use the Debug build when available and building in Debug configuration.
-    -->
-    <BlazorServerJSFile>..\..\Web.JS\dist\Release\blazor.server.js</BlazorServerJSFile>
-    <BlazorServerJSFile Condition="'$(Configuration)' == 'Debug' AND Exists('..\..\Web.JS\dist\Debug\blazor.server.js')">..\..\Web.JS\dist\Debug\blazor.server.js</BlazorServerJSFile>
+    <BlazorServerJSFile>..\..\Web.JS\dist\$(Configuration)\blazor.server.js</BlazorServerJSFile>
   </PropertyGroup>
 
   <ItemGroup>
-    <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/%(Filename)%(Extension)" />
+    <EmbeddedResource Include="$(BlazorServerJSFile)" LogicalName="_framework/%(Filename)%(Extension)" Condition="Exists('$(BlazorServerJSFile)')" />
```

Comment:
This is really dangerous. It's not valid to build this assembly without the embedded resource, and if it were to regress it would be time consuming to investigate. 

I consider this case to be a silent failure. It is a hard requirement that we build the .js files in order to build this assembly.

Reply:
That's why I added a VerifyBlazorServerJSFileExists target that runs before build. This will fail release builds if the files are not present but will only warn for debug builds. Is that sufficient?

Reply:
I'd really prefer it be an error in both cases.

Reply:
Then we'll need to build NodeJS on all builds (where this project is built) which answers https://github.com/aspnet/AspNetCore/pull/12024#discussion_r302672574 too.